### ISSUE TITLE
Don't override keep_message value in the outer prefix_stack_trace-marked functions

### ISSUE THREAD

**Issue by evgri243** [Azure/shrike#160]
**Category**
Please select:
- [ ] shrike.pipeline
- [ ] shrike.build
- [x] shrike.compliant_logging

**Is your feature request related to a problem? Please describe.**
Apparently, the outer `prefix_stack_trace` applies its own `keep_message` value on the exception, even if it was raised from a function marked as "safe" with `keep_message=True`. So that, it scrubs error message even if it comes from safe calls.

Example:
```python
@prefix_stack_trace(keep_message=True)
def inner_safe_metadata_only_function():
    raise ValueError("This message is safe")

@prefix_stack_trace(keep_message=False)
def outer_dangerous_function_working_with_data():
    save_column_only_function()
    
    if 1==1:
        raise ValueError(f"User data: {data}")

outer_dangerous_function_working_with_data()
```
In that scenario the message will be scrubbed.

**Describe the solution you'd like**
There is a beautiful pattern allowing the function to decide itself if it can produce dangerous non-compliant messages on its own. Alternatively, we can get all the safe input from the orchestrating function, but still keep unsafe errors scrubbed.

```python
@prefix_stack_trace(keep_message=False)
def inner_dangerous_function_working_with_data():
    raise ValueError(f"User data: {data}")

@prefix_stack_trace(keep_message=True)
def outher_safe_orchestrating_only_function():
    inner_dangerous_function_working_with_data()
    
    if 1==1:
         raise ValueError("This message is safe")

outher_safe_orchestrating_only_function()
```

**Describe alternatives you've considered**
Keep as is. The current solution is strict and forces developers to use a complicated function decomposition to achieve a similar effect.
Make it a flag to make this logic configurable.

**Estimate the Impact**
1. How many people/teams would benefit from that new feature?
    Up to all shrike.compliant_logging users, but hard to say exactly. Depends on how this pattern is about to be advertised.

2. Are you able to quantify the impact of (not) having this feature? E.g. "this would save me 2 hours a week" or "not having this functionality makes us use 15% more CPU time". 
     Low priority convenience improvement.

**Comment by dkmiller** [comment]
@evgri243 , thanks for raising the issue!

The ask generally makes sense, but implementation could be subtle. We'll spend some time discussing this in detail next week and get back to you. In parallel, if you have implementation ideas, please don't hesitate to share them in the discussion for this issue.

**Comment by evgri243** [comment]
I haven't yet deeply researched your exception handling code, but one of the ideas was to use python's `setattr` function to set something like "__shike_visited__" on the exception value at the closest @prefix_stack_tract, prepare it there and don't touch it any more through the stack trace.

**Comment by dkmiller** [comment]
@evgri243 excellent suggestion.

We'll have to think carefully about this since changes to scrubbing logic carry a risk of leaking customer content, but this gives plenty to go on.

**Comment by dkmiller** [comment]
Consensus: let's do this, but not committing right away to a specific timeline.

Alternatively, if @evgri243 wants it quickly, you're welcome to contribute :) .
